tests/register/register.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package register
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/init/tests/util"
+)
+
+type Test struct {
+	Name            string
+	Func            func(*testing.T, Test)
+	BinaryPath      string
+	IgnitionConfig  *string
+	CloudinitConfig *string
+}
+
+func (test Test) Run(t *testing.T) {
+	if os.Getenv("TMPDIR") == "" {
+		tmpDir, err := ioutil.TempDir("/var/tmp", "")
+		if err != nil {
+			t.Fatalf("failed to create temp working dir in /var/tmp: %v", err)
+		}
+
+		err = os.Setenv("TMPDIR", tmpDir)
+		if err != nil {
+			t.Fatalf("couldn't set TMPDIR env var: %v", err)
+		}
+
+		defer test.RemoveAll(t, tmpDir)
+		defer os.Setenv("TMPDIR", "")
+	}
+	test.Func(t, test)
+}
+
+func (test Test) CreateDevice(t *testing.T) (string, string) {
+	diskFile, err := ioutil.TempFile("", "coreos-install-disk")
+	if err != nil {
+		t.Fatalf("failed to create disk file: %v", err)
+	}
+
+	// truncate the disk file to 10GB, this should be large enough
+	err = os.Truncate(diskFile.Name(), 10*1024*1024*1024)
+	if err != nil {
+		t.Fatalf("failed to truncate disk file: %v", err)
+	}
+
+	// back a loop device with the disk file
+	device := string(util.MustRun(t, "losetup", "-P", "-f", diskFile.Name(), "--show"))
+	return diskFile.Name(), strings.TrimSpace(device)
+}
+
+func (test Test) CleanupDisk(t *testing.T, diskFile, loopDevice string) {
+	util.MustRun(t, "losetup", "-d", loopDevice)
+}
+
+func (test Test) MountPartitions(t *testing.T, loopDevice string) (string, string) {
+	efi, err := ioutil.TempDir("", "efi-mount-point")
```

Comment:
This is normally mounted on `/boot`, so you can save yourself the extra mount point.